_includes/specials/taproot/en/01-single-sig.md

```diff
@@ -0,0 +1,75 @@
+using P2PKH inputs and outputs---but, perhaps surprisingly, P2TR
+transactions are slightly larger than equivalent P2WPKH transactions.
+
+|                    | P2PKH (legacy) | P2WPKH (segwit v0) | P2TR (taproot/segwit v1) |
+|--------------------|----------------|--------------------|--------------------------|
+| **Output**         |  34            | 31                 | 43                       |
+| **Input**          |  148           | 67.75              | 57.25                    |
+| **2-in, 2-out tx** |  374           | 208.25             | 211.25                   |
+
+That may make it seem counterproductive for single-sig wallets to
+implement taproot spending in preparation for block {{site.trb}}, but a
+closer look reveals that there are a number of advantages to using P2TR
+for single-sigs, both for wallet users and for the network as a whole.
+
+- **Cheaper to spend:** it costs about 15% less at the input level to
+  spend a single-sig P2TR UTXO than it does to spend a P2WPKH UTXO.  An
+  overly simple analysis like the table above hides the detail that you
+  can't choose which addresses you're asked to pay, so if you stay on
+  P2WPKH and everyone else upgrades to P2TR, the actual typical size of
+  your 2-in-2-out transactions will be 232.25 vbytes---while all-P2TR
+  transactions will still only be 211.25 vbytes.
+
+- **Privacy:** although some privacy is lost when early adopters change
+  to a new script format, users switching to taproot also immediately
+  receive a privacy boost.  Your transactions will be able to look
+  indistinguishable from people working on new LN channels, more
+  efficient [DLCs][topic dlc], secure [multisignatures][topic
+  multisignature], various clever wallet backup recovery schemes, or a
+  hundred other pioneering developments.
+
+    Using P2TR for single-sig now also allows your wallet to upgrade to
+    multisignatures, tapscripts, LN support, or other features later on
+    without affecting the privacy of your existing users.  It won't
+    matter whether a UTXO was received to an old version or a new
+    version of your software---both UTXOs will look the same onchain.
+
+- **More convenient for hardware signing devices:** since the
+  rediscovery of the [fee overpayment attack][news101 fee overpayment
+  attack], several hardware signing devices have refused to sign a
+  transaction unless each UTXO spent in that transaction is accompanied
+  by metadata containing a copy of significant parts of the entire
+  transaction which created that UTXO.  This greatly increases the
+  worst-case processing that hardware signers need to perform and is
+  especially problematic for hardware signers using limited-size QR
+  codes as their primary communication medium.  Taproot eliminates the
+  vulnerability underlying the fee overpayment attack and so can
+  significantly improve the performance of hardware signers.
+
+- **More predictable feerates:** ECDSA signatures for P2PKH and P2WPKH
```

Comment:
Grinding increases the likelihood that your predicted feerate will be your obtained feerate, but it doesn't guarantee it, since you could always end up with an even smaller R than you expected (e.g. you have a `1 / 2 / 256` chance at a 70-byte signature, a `1 / 2 / 256^2` chance at 69 bytes, etc..).

Additionally, while I think signature grinding is probably overall a good idea, it also feels to me a bit like rolling your own crypto.  Andrew Chow's PR to grind in Bitcoin Core had review by experienced cryptographers; but I wouldn't blame a cautious solo wallet dev about being reluctant to do something that might risk introducing bias into his signatures or otherwise risking problems the dev didn't fully understand.

Reply:
As far as I'm aware issues with incorrect fees are caused by underpaying fees (like conflicting with the `minRelayTxFeeRate`, or Bitpay's inane minimum fees), so getting a smaller `r` simply increases the feerate of your transaction beyond the intended, which happens all the time with conservative estimates anyway.

The "roll your own" crypto makes sense to me, though, although, I would suspect that most wallets use some standard library like `libsecp`, or `bitcoinjs`.